### PR TITLE
update loss.py in nn.functional and nn.layer module;test=document_fix 

### PR DESCRIPTION
**Pull request by liyongchao911**
### PR types
Others

### PR changes
Docs

### Describe
1. fix some typo and markdown issue
2. add optional note in the nn.Pad1D/2D/3D class parameter, for both CN and EN docs
3. replace np.prod with paddle.prod in some code example

### CODE REVIEW

File: python/paddle/nn/layer/common.py [hunk 1]
```diff
@@ -954,43 +954,31 @@ class Pad1D(Layer):
...
-
-            x = [[[1., 2., 3.],
-                  [4., 5., 6.]]]
-            padding = [1, 2],
-            mode = "constant"
-            value = 0.0
-            Out = [[[0. 1. 2. 3. 0. 0.]
-                    [0. 4. 5. 6. 0. 0.]]]
-
     Code Examples:
```
Comment (SigureMo) [inline python/paddle/nn/layer/common.py]: 这里改成 `Examples:` 吧～

**Review by SigureMo — CHANGES_REQUESTED** [review]

**Comment by SigureMo** [comment]
~~最好中英互相链起来呀～~~ 我蠢了，这俩 PR 好像没有对应关系 :joy:

File: python/paddle/nn/layer/common.py (hunk 1 above)
Comment (liyongchao911) [reply python/paddle/nn/layer/common.py]: Done！

File: python/paddle/nn/functional/common.py [hunk 2]
```diff
@@ -1317,25 +1317,23 @@ def pad(x, pad, mode='constant', value=0, data_format="NCHW", name=None):
...
-            When in 'reflect' mode, uses reflection of the input boundaries to pad the input tensor.
-            When in 'replicate' mode, uses input boundaries to pad the input tensor.
-            When in 'circular' mode, uses circular input to pad the input tensor.
+           - 'constant' mode, uses a constant value to pad the input tensor.
+           - 'reflect' mode, uses reflection of the input boundaries to pad the input tensor.
+           - 'replicate' mode, uses input boundaries to pad the input tensor.
+           - 'circular' mode, uses circular input to pad the input tensor.
             Default is 'constant'
-        value (float32, optional): The value to fill the padded areas in 'constant' mode . Default is 0.0
+        value (float, optional): The value to fill the padded areas in 'constant' mode . Default is 0.0
```
Comment (SigureMo) [inline python/paddle/nn/functional/common.py]: 这里和中文对齐吧，同样用

```
:math:`0.0`
```

最后也需要加英文句号 `.`

File: python/paddle/nn/functional/common.py [hunk 3]
```diff
@@ -1317,25 +1317,23 @@ def pad(x, pad, mode='constant', value=0, data_format="NCHW", name=None):
...
         mode (str, optional): Four modes: 'constant' (default), 'reflect', 'replicate', 'circular'.
-            When in 'constant' mode, this op uses a constant value to pad the input tensor.
-            When in 'reflect' mode, uses reflection of the input boundaries to pad the input tensor.
-            When in 'replicate' mode, uses input boundaries to pad the input tensor.
-            When in 'circular' mode, uses circular input to pad the input tensor.
+           - 'constant' mode, uses a constant value to pad the input tensor.
+           - 'reflect' mode, uses reflection of the input boundaries to pad the input tensor.
+           - 'replicate' mode, uses input boundaries to pad the input tensor.
+           - 'circular' mode, uses circular input to pad the input tensor.
             Default is 'constant'
```
Comment (SigureMo) [inline python/paddle/nn/functional/common.py]: 这里默认值和英文调整成一样的吧，都放到上面去，此外同样空一下行以免渲染问题吧

File: python/paddle/nn/functional/loss.py [hunk 4]
```diff
@@ -1316,7 +1316,7 @@ def nll_loss(input,
              it treated as if having all ones. the data type is
              float32, float64, Default is ``'None'``.
          ignore_index (int64, optional): Specifies a target value that is ignored
```
Comment (SigureMo) [inline python/paddle/nn/functional/loss.py]: 这里 `int64` 改为 `int`

**Review by SigureMo — CHANGES_REQUESTED** [review]

File: python/paddle/nn/functional/loss.py (hunk 4 above)
Comment (Ligoml) [reply python/paddle/nn/functional/loss.py]: 这里好像还是没改？

File: python/paddle/nn/layer/loss.py [hunk 5]
```diff
@@ -835,16 +835,15 @@ class NLLLoss(Layer):
...
             If `reduction` is ``'mean'``, the reduced mean loss is returned;
             if `reduction` is ``'sum'``, the reduced sum loss is returned;
             if `reduction` is ``'none'``, no reduction will be apllied.
             Default is ``'mean'``.
-         name (str, optional): Name for the operation (optional, default is None).
-             For more information, please refer to :ref:`api_guide_Name`.
+         name (str, optional): For details, please refer to :ref:`api_guide_Name`. Generally, no setting is required. Default: None.
 
     Shape:
         input (Tensor): Input tensor, the shape is :math:`[N, C]`, `C` is the number of classes.
```
Comment (Ligoml) [inline python/paddle/nn/layer/loss.py]: image
这里需要增加空行，保证前端正确展示

File: python/paddle/nn/layer/loss.py [hunk 6]
```diff
@@ -835,16 +835,15 @@ class NLLLoss(Layer):
...
             float32, float64, Default is ``'None'``.
-        ignore_index (int64, optional): Specifies a target value that is ignored
+        ignore_index (int, optional): Specifies a target value that is ignored
             and does not contribute to the input gradient.
         reduction (str, optional): Indicate how to average the loss,
             the candicates are ``'none'`` | ``'mean'`` | ``'sum'``.
             If `reduction` is ``'mean'``, the reduced mean loss is returned;
             if `reduction` is ``'sum'``, the reduced sum loss is returned;
             if `reduction` is ``'none'``, no reduction will be apllied.
             Default is ``'mean'``.
```
Comment (Ligoml) [inline python/paddle/nn/layer/loss.py:845]: image
对应的中文文档似乎增加了多余的符号

**Review by Ligoml — DISMISSED** [review]

File: python/paddle/nn/layer/loss.py (hunk 5 above)
Comment (Ligoml) [reply python/paddle/nn/layer/loss.py]: 抱歉我记错了，这里是需要在前面加`-`，我直接改了

**Review by SigureMo — APPROVED** [review]
LGTM~~~

**Review by Ligoml — APPROVED** [review]
LGTM
